test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java

```diff
@@ -166,7 +166,7 @@
 import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.lessThanOrEqualTo;
 
-public abstract class InternalAggregationTestCase<T extends InternalAggregation> extends AbstractWireSerializingTestCase<T> {
+public abstract class InternalAggregationTestCase<T extends InternalAggregation> extends AbstractNamedWriteableTestCase<T> {
```

Comment:
I think it is worth adding a comment that subclasses need to make sure that `xContentRegistry` includes the agg we're going to parse. Built in aggs are included automatically but other aggs won't be.

Reply:
Buncha xpack tests needed this treatment, and it was non-trivial for tests to convert aggspecs into namedwriteables, so I did a bit of refactoring to make it easier in the general case.